reporting-anything/src/download_helpers.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import puppeteer from 'puppeteer-core';
+import createDOMPurify from 'dompurify';
+import fs from 'fs';
+import { JSDOM } from 'jsdom';
+import { CHROMIUM_PATH, REPORT_TYPE_URLS, FORMAT, REPORT_TYPE, SELECTOR } from './constants.js';
+
+
+export async function downloadVisualReport(url, type, object_id, format, width, height) {
+  const window = new JSDOM('').window;
+  const DOMPurify = createDOMPurify(window);
+
+  try {
+    const browser = await puppeteer.launch({
+      headless: true,
+      /**
+       * TODO: temp fix to disable sandbox when launching chromium on Linux instance
+       * https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#setting-up-chrome-linux-sandbox
+       */
+      args: [
+        '--no-sandbox',
+        '--disable-setuid-sandbox',
+        '--disable-gpu',
+        '--no-zygote',
+        '--single-process',
+        '--font-render-hinting=none',
+      ],
+      executablePath: CHROMIUM_PATH,
+      ignoreHTTPSErrors: true,
+      env: {
+        TZ: 'UTC', // leave as UTC for now
+      },
+    });
+
+    const page = await browser.newPage();
+    page.setDefaultNavigationTimeout(0);
+    page.setDefaultTimeout(100000);
+
+    let queryUrl = '';
+    if (url != undefined) {
+      queryUrl = url;
+    }
+    else {
+      queryUrl = 'http://localhost:5601/app/';
+      // create URL from report source type and saved object ID
+      if (type.toUpperCase() === 'DASHBOARD') {
+        queryUrl += REPORT_TYPE_URLS.DASHBOARD;
+      }
+      else if (type.toUpperCase() === 'VISUALIZATION') {
+        queryUrl += REPORT_TYPE_URLS.VISUALIZATION;
+      }
+      else if (type.toUpperCase() === 'NOTEBOOK') {
+        queryUrl += REPORT_TYPE_URLS.NOTEBOOK;
+      }
+      queryUrl += '/' + object_id;
+    }
+    await page.goto(queryUrl, { waitUntil: 'networkidle0' });
```

Comment:
how is auth handled?

Reply:
One of the goals of Reporting Anything is that auth issues should be handled from the customer side. If they run the CLI from their own workflow/env, they should not encounter any auth issues

Reply:
any instructions on how that should be done? 

e.g. i'm a user who has a security enabled domain, how do i handle auth on my side so reporting anything can use my credentials? or "handled from the customer side" means disabling auth?

Reply:
It depends on use case- but for your example, "handled from the customer side" to me could mean using reporting anything in a lambda workflow; if my understanding of the service is correct, it would use the same security credentials as the OpenSearch domain. From there, the customer can download the reports they would like, and integrate it further with an email service to create a workaround for email reports.

Reply:
i meant if i have a self hosted security domain. And i don't think it happens automatically even for your case, how would you specify different user/tenants? those are opensearch specific concepts. It's ok if initial PR doesn't support security, but if we are claiming auth can work we should at least test it first (if you haven't done so yet)